File: Leetcode/5 - Longest Palindromic Substring/solution.py
def longestPalindrome(self, s: str) -> str:
    def get_palindrome_bounds(low: int, high: int) -> Tuple[int, int]:
        while low >= 0 and high < len(s) and s[low] == s[high]:
            low -= 1
            high += 1
        return low + 1, high

    max_start: int = 0
    max_end: int = 1

    for i in range(len(s)):
        start1, end1 = get_palindrome_bounds(i, i)         # odd-length center
        if end1 - start1 > max_end - max_start:
            max_start, max_end = start1, end1

        start2, end2 = get_palindrome_bounds(i, i + 1)     # even-length center
        if end2 - start2 > max_end - max_start:
            max_start, max_end = start2, end2

    return s[max_start:max_end]

File: Leetcode/5 - Longest Palindromic Substring/solution.js
var longestPalindrome = function (s) {
    const getPalindromeLength = (low, high) => {
        while (low >= 0 && high < s.length && s[low] === s[high]) {
            low--;
            high++;
        }
        return [low + 1, high];
    };

    let maxStart = 0, maxEnd = 1;
    for (let i = 0; i < s.length; i++) {
        let [start1, end1] = getPalindromeLength(i, i);
        if (end1 - start1 > maxEnd - maxStart) {
            maxStart = start1;
            maxEnd = end1;
        }
        let [start2, end2] = getPalindromeLength(i, i + 1);
        if (end2 - start2 > maxEnd - maxStart) {
            maxStart = start2;
            maxEnd = end2;
        }
    }

    return s.slice(maxStart, maxEnd);
};